### PR TITLE
[CEN-1020] Pipeline code review

### PR DESCRIPTION
**Pull request by GiovanniMancini**
## Description
This PR proposes to add a code review pipeline (Azure DevOps) which can be executed on PR.
The pipeline send JaCoCo reports to SonarCloud for analysis. Test for correct handling of differential metrics, like line and branch coverage, is to be postponed to other PRs

### CODE REVIEW

File: .devops/code-review-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+trigger:
+  - master
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+#variables:
```
Comment (alberto-re-pagopa) [inline .devops/code-review-pipelines.yml:7]: Why commit commented lines? If there is no need for those instructions simply delete them

File: .devops/code-review-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      SonarCloud: '$(SONARCLOUD_SERVICE_CONN)'
+      organization: '$(SONARCLOUD_ORG)'
+      scannerMode: Other
+      extraProperties: |
+        sonar.projectKey=$(SONARCLOUD_PROJECT_KEY)
+        sonar.projectName=$(SONARCLOUD_PROJECT_NAME)
+        sonar.coverage.jacoco.xmlReportPaths=**/target/site/jacoco-aggregate/jacoco.xml
+        sonar.junit.reportPaths=**/target/surefire-reports/
+        sonar.exclusions='**/enums/**, **/model/**, **/*Constant*, **/*Config.java, **/*Scheduler.java, **/*Application.java, **/src/test/**, **/Dummy*.java'
+  # - task: Maven@3
```
Comment (alberto-re-pagopa) [inline .devops/code-review-pipelines.yml:25]: Why commit commented lines? If there is no need for those instructions simply delete them

File: .devops/code-review-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      publishJUnitResults: false
+      javaHomeOption: 'JDKVersion'
+      mavenVersionOption: 'Default'
+      mavenAuthenticateFeed: false
+      effectivePomSkip: false
+      sonarQubeRunAnalysis: true
+      isJacocoCoverageReportXML: true
+      testResultsFiles: '**/surefire-reports/TEST-*.xml'
+
+      # sqMavenPluginVersionChoice: 'latest'
```
Comment (alberto-re-pagopa) [inline .devops/code-review-pipelines.yml:52]: Why commit commented lines? If there is no need for those instructions simply delete them

**Review by alberto-re-pagopa — CHANGES_REQUESTED** [review]
Except for the lines commented looks good to me

File: .devops/code-review-pipelines.yml (hunk 3 above)
Comment (GiovanniMancini) [reply .devops/code-review-pipelines.yml:52]: Thank you, I'll fix

**Review by alberto-re-pagopa — APPROVED** [review]

**Pull request merged**